### PR TITLE
Clean up

### PR DESCRIPTION
**Pull request by s-saleem**

### CODE REVIEW

File: common.py [hunk 1]
```diff
@@ -43,26 +44,26 @@ def load_d4rl_er(h5path, batch_size, history_length, traj_length):
...
+    flattened_post_states = np.roll(flattened_states, -1, axis=0)
     flattened_post_states[-1] = flattened_post_states[-2] # the last post-state uses the pre-state
     terminals = data_dict["terminals"]
-    inverted_terminals= terminals.invert()
+    inverted_terminals= np.invert(terminals)
     # masked out other states, only keep terminal states
     terminal_post_states = np.ma.masked_array(
         flattened_states,
-        mask=np.column_stack([inverted_terminals for _ in flattened_post_states.shape[-1]]),
+        mask=np.column_stack([inverted_terminals for _ in range(flattened_post_states.shape[-1])]),
```
Comment (lalitlal) [inline common.py:53]: Could you explain why this change was needed ?

File: environment.py [hunk 2]
```diff
@@ -18,14 +19,14 @@ def _step(self, action):
         result = self.gym.step(action)
         self.state, self.reward, self.done, self.info = result[:4]
         if self.random_initialization:
-            return np.float32(self.state), np.float32(self.reward), self.done
+            return np.float32(self.state['image']).reshape(1,-1), np.float32(self.reward), self.done
```
Comment (lalitlal) [inline environment.py:21]: Is it just me or there's no different  between the code in the if statement or the else statement?

File: common.py (hunk 1 above)
Comment (s-saleem) [reply common.py:53]: yea this line is just creating a history_size X state_dim matrix that's sets a row to True if that state is NOT a terminal state. This masks out those states from flattened_states.

Just needed the range() for the for loop  to stack `inverted_terminals` array `flattened_post_states.shape[-1]` (= state_dim)  times

File: environment.py (hunk 2 above)
Comment (s-saleem) [reply environment.py:21]: Yea these don't seem to do anything different, I couldn't find how this was different before. This may need to be looked into still.

Comment (yipsang) [reply environment.py:21]: https://github.com/itaicaspi/mgail/blob/master/environment.py#L20-L24

When there were qpos and qvel, there is difference, but I think this condition can be removed now

File: driver.py [hunk 3]
```diff
@@ -103,8 +103,8 @@ def collect_experience(self, record=1, vis=0, n_steps=None, noise_flag=True, sta
             observation = self.env.reset()
             observation = observation['image']
         else:
-            qposs, qvels = alg.er_expert.sample()[5:]
-            observation = self.env.reset(qpos=qposs[0], qvel=qvels[0])
+            states, actions, rewards, posstates, terminals = alg.er_expert.sample()
+            observation = states[5]
```
Comment (yipsang) [inline driver.py:106]: Why are we picking particularly the 5th entry of the states?

Comment (s-saleem) [reply driver.py:106]: it seems like the old code picks the 5th entry too here `alg.er_expert.sample()[5:]`? I assumed because the first few states maybe unreliable somehow?
Should i change it to 0?

Comment (yipsang) [reply driver.py:106]: the old code `[5:]` is just picking the values starting from the 5th entry which are originally qpos and qvels of the tuple from `sample()`

Comment (yipsang) [reply driver.py:106]: yes, I think states[0] should be fine, it just takes the 1st state in the original code, just it recovers it from qpos and qvel
